internal/datastore/common/errors.go

```diff
@@ -0,0 +1,15 @@
+package common
+
+import (
+	"context"
+
+	"github.com/rs/zerolog/log"
+)
+
+// LogOnError executes the function and logs the error.
+// Useful to avoid silently ignoring errors in defer statements
+func LogOnError(ctx context.Context, f func() error) {
+	if err := f(); err != nil {
+		log.Ctx(ctx).Error().Err(err)
```

Comment:
this is pulling the logger from the context, but in both call sites we're calling `LogOnError` with an empty context. Since we don't set the [DefaultContextLogger](https://github.com/rs/zerolog/blob/c0c2e11fc3cd04ae28d856789cf58ffd1666bc3f/globals.go#L106), this is likely swallowing the error rather than logging it

Reply:
Ah, that's a good call out!  I mentioned this is part of a larger set of changes that we've got in our internal fork.  Most other call sites in the change _do_ pass a context with an inject logger, but these two happen to not.

I don't think it's appropriate to set a `DefaultContextLogger` within the `mysql` or `migrations` package here as it would modify logging for the whole application. I've opted to inject the [default `log.Logger`](https://pkg.go.dev/github.com/rs/zerolog/log#pkg-variables) into the context in these two call spots.  [c6ab86b](https://github.com/authzed/spicedb/pull/411/commits/c6ab86bd42cbde03526837cab83bd50cf7fbe6a3)

Thoughts?

Reply:
That seems reasonable for now 👍 

We should probably (separately) track an effort to make the use of a context logger consistent throughout the codebase